surge/blueprints.py

```diff
@@ -0,0 +1,94 @@
+import re
+from surge.projects import Project
+from surge.errors import SurgeMissingAttributeError
+
+
+class Blueprint(Project):
+    def __int__(self, **kwargs):
+        super().__init__(kwargs)
+
+    @classmethod
+    def _from_project(cls, project):
+        '''Converts a Project to a Blueprint. This is useful when we want to create a Project and then return
+        a Blueprint to the user.
+
+        Arguments:
+            project (Project): The project to convert to a Blueprint
+
+        Returns:
+            blueprint (Blueprint): A Blueprint derived from the given Project
+        '''
+        kwargs = project.__dict__
+        # A Project object will have already parsed created_at, so omit that and save current value to avoid errors.
+        created_at = kwargs.pop('created_at', None)
+        b = Blueprint(**project.__dict__)
+        b.created_at = created_at
+        return b
+
+    def required_data_fields(self):
```

Comment:
This probably is outside the scope here but is this validated on the server side? I think we have fairly elaborate templating logic that could continue to evolve so this probably won't be exhaustive